pkgs/desktops/cinnamon/cjs/default.nix

```diff
@@ -17,26 +15,51 @@
 , libffi
 , gtk3
 , readline
+, spidermonkey_78
+, meson
+, sysprof
+, dbus
+, xvfb_run
+, ninja
+, makeWrapper
+, which
+, libxml2
 }:
 
-let
-
-  # https://github.com/linuxmint/cjs/issues/80
-  spidermonkey_52 = callPackage ./spidermonkey_52.nix {};
-
-in
-
 stdenv.mkDerivation rec {
   pname = "cjs";
-  version = "4.6.0";
+  version = "4.6.0-gjs-1.66.0";
 
   src = fetchFromGitHub {
-    owner = "linuxmint";
+    owner = "leigh123linux";
```

Comment:
What's wrong with upstream?

Reply:
You should grab https://github.com/linuxmint/cjs/pull/84 as a patch.

Reply:
The patch is quite gigantic. There have been some problems with renames not being applied in previous patches I applied, so I decided to use that approach as it seems saner.